### PR TITLE
Frontend refactor

### PR DESCRIPTION
**Pull request by brandoncorey**
A general refactor of the frontend. Notable changes are:
1. Moving drop down data into a `data/dropdowns/` directory. This removes a lot of clutter from the code, but there may be a better way to do this.
2. Introduce `src/services` directory. This puts all of the axios calls into their own modules, making for a cleaner interface when making network calls
3. Introduce `src/reducers` and `src/hooks`, which implement
a. A reducer that is used to manage the state of dynamic queries to the backend made in the query builder
b. A custom hook `useQueryReducer` that is used to dispatch actions and update the state of the current query

### CODE REVIEW

File: data/dates/ranges.json
Comment (AlliLearns) [inline data/dates/ranges.json:1]: Thanks for adding a default!

File: src/components/DateButton.jsx
Comment (AlliLearns) [inline src/components/DateButton.jsx:1]: Please remove the commented `console.log` (if you don't need it anymore).

File: src/components/QueryBuilder.jsx
Comment (AlliLearns) [inline src/components/QueryBuilder.jsx:1]: There's a typo with `selecteDataRange` state. Should be `selectedDateRange`.

**Review by AlliLearns — CHANGES_REQUESTED** [review]
- I really like the addition of the `data` directory.
- Add comments to appropriate files in the `data` directory that lets the reader know that if a value changes, a corresponding value must be changed on the backend repository.
- I also appreciate the variable name changes.
- Potentially replace multi-line destructuring in component parameter lists and move the destructuring to within the component. (`DateRange`, `Dropdown`, `FilterDropdown`, `FilterDropdownOptions` and `Metric`)
- We should remove the `ResultTable` component if we're not going to implement it.

File: src/components/QueryBuilder.jsx
Comment (brandoncorey) [reply src/components/QueryBuilder.jsx:1]: This was a very good catch wow

**Comment by brandoncorey** [comment]
So I made the relevant changes except for two:
1. I did not add the comments for the data file as JSON doesn't seem to support this, so we may need to include the info in our docs and README
2. The destructuring within the components is more of a style change, and it might be better to hop on a call about this one. If so, we can put the resolution of that conversation in this thread.

**Comment by brandoncorey** [comment]
Allison and I discussed the style changes are are leaving the destructured props as is for now

**Review by AlliLearns — APPROVED** [review]
Looks good, thanks!

**Pull request merged**